core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroupSetters.java

```diff
@@ -16,19 +16,45 @@
 
 package com.linecorp.armeria.client.endpoint;
 
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
 import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.Flags;
 import com.linecorp.armeria.common.annotation.UnstableApi;
 
 /**
  * Sets properties for building {@link DynamicEndpointGroup}.
  */
 @UnstableApi
-@SuppressWarnings("InterfaceMayBeAnnotatedFunctional")
 public interface DynamicEndpointGroupSetters {
 
     /**
      * Sets whether to allow an empty {@link Endpoint} list.
      * If unspecified, an empty {@link Endpoint} list is allowed.
      */
     DynamicEndpointGroupSetters allowEmptyEndpoints(boolean allowEmptyEndpoints);
+
+    /**
+     * Sets the timeout to wait until a successful {@link Endpoint} selection.
+     * {@link Duration#ZERO} disables the timeout.
+     * If unspecified, {@link Flags#defaultConnectTimeoutMillis()} is used by default.
+     */
+    @UnstableApi
+    default DynamicEndpointGroupSetters selectionTimeout(Duration selectionTimeout) {
```

Comment:
`selectionTimeout` or `selectTimeout`? I feel like `selectionTimeout` is correct, but we have `connectTimeout` 🤔

Reply:
Probably `selectionTimeout` is better here because we have `EndpointSelectionTimeoutException`.